### PR TITLE
Prep script and Fixes for single shot detector (SSD)

### PR DESCRIPTION
**Pull request by larroy**
## Description ##
Fix multibox detector kernel resource limit on launch on volta using __launch_bounds__

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by larroy** [comment]
@KellenSunderland

**Comment by larroy** [comment]
@eric-haibin-lin

File: example/ssd/data/demo/download_demo_images.py [hunk 1]
```diff
@@ -1,3 +1,6 @@
+#!/usr/bin/env python3
```
Comment (KellenSunderland) [inline example/ssd/data/demo/download_demo_images.py]: Do we have a standard for headers?  It'd be great if we could follow a convention for what to put here.

Comment (larroy) [reply example/ssd/data/demo/download_demo_images.py]: Would be great. As far as I know this is the recommended best practice.

**Comment by larroy** [comment]
@zhreshold

File: src/operator/contrib/multibox_detection.cu [hunk 2]
```diff
@@ -51,7 +51,9 @@ __device__ void CalculateOverlap(const DType *a, const DType *b, DType *iou) {
 }
 
 template<typename DType>
-__global__ void DetectionForwardKernel(DType *out, const DType *cls_prob,
+__global__
+__launch_bounds__(cuda::kMaxThreadsPerBlock)
```
Comment (zhreshold) [inline src/operator/contrib/multibox_detection.cu]: Can you explain why Volta need this?

Comment (larroy) [reply src/operator/contrib/multibox_detection.cu]: Kernel will fail to launch, too many resources.

**Pull request merged**